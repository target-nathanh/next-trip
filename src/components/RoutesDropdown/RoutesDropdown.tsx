import React from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import { Route } from '../../types';
import { useDropdownStyles } from '../../common/styles/dropDownStyles';

interface RoutesDropdownProps {
  routes?: Route[];
  selectedRouteId?: string;
  onSelectRoute: (routeId: string) => void;
}

const RoutesDropdown: React.FC<RoutesDropdownProps> = ({
  routes,
  onSelectRoute,
  selectedRouteId,
}) => {
  const classes = useDropdownStyles();

  return (
    <Select
      className={classes.select}
      value={selectedRouteId}
      placeholder={'Select route'}
      onChange={(event: React.ChangeEvent<{ value: unknown }>) =>
        onSelectRoute(event.target.value as string)
      }
    >
      {routes?.map((route: Route) => (
        <MenuItem key={route.route_id} value={route.route_id}>
          {route.route_label}
        </MenuItem>
      ))}
    </Select>
  );
};

export default RoutesDropdown;
