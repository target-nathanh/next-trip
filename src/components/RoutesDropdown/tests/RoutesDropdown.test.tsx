import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import routes from '../../../mockData/routesMock.json';
import RoutesDropdown from '../RoutesDropdown';
import { MemoryRouter } from 'react-router-dom';

describe('RoutesDropdown', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <RoutesDropdown
          routes={routes}
          selectedRouteId={routes[0].route_id}
          onSelectRoute={() => {}}
        />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('calls onSelectRoute when new route is selected', async () => {
    const onSelectRouteMock = jest.fn();
    const { getByRole } = render(
      <MemoryRouter>
        <RoutesDropdown
          routes={routes}
          selectedRouteId={routes[0].route_id}
          onSelectRoute={onSelectRouteMock}
        />
      </MemoryRouter>
    );
    fireEvent.mouseDown(getByRole('button'));

    const option = await screen.findByText(routes[1].route_label);
    fireEvent.click(option);
    expect(onSelectRouteMock).toBeCalledWith(routes[1].route_id);
  });
});
