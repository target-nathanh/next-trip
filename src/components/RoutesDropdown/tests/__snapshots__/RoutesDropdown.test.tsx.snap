// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutesDropdown matches snapshot 1`] = `
<div
  class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-1"
>
  <div
    aria-haspopup="listbox"
    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
    role="button"
    tabindex="0"
  >
    METRO Blue Line
  </div>
  <input
    aria-hidden="true"
    class="MuiSelect-nativeInput"
    placeholder="Select route"
    tabindex="-1"
    value="901"
  />
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSelect-icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;
